Trying to add a disposable to a DisposableStore that has already been disposed of

This happens when reloading the window with a longish codeblock in the history

```
lifecycle.js:329 Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!
    at DisposableStore.add (lifecycle.js:329:34)
    at ChatTextEditContentPart._register (lifecycle.js:393:32)
    at chatTextEditContentPart.js:104:30
    at async CodeCompareBlockPart.updateEditor (codeBlockPart.js:571:30)
    at async CodeCompareBlockPart.render (codeBlockPart.js:526:13)
```

--- a/src/vs/workbench/contrib/chat/browser/codeBlockPart.ts
+++ b/src/vs/workbench/contrib/chat/browser/codeBlockPart.ts
@@ -144,6 +144,7 @@ export class CodeBlockPart extends Disposable {
 	private currentScrollWidth = 0;
 
 	private readonly disposableStore = this._register(new DisposableStore());
+	private isDisposed = false;
 
 	constructor(
 		private readonly options: ChatEditorOptions,
@@ -264,6 +265,11 @@ export class CodeBlockPart extends Disposable {
 		}
 	}
 
+	override dispose() {
+		this.isDisposed = true;
+		super.dispose();
+	}
+
 	get uri(): URI | undefined {
 		return this.editor.getModel()?.uri;
 	}
@@ -356,6 +362,9 @@ export class CodeBlockPart extends Disposable {
 		}
 
 		await this.updateEditor(data);
+		if (this.isDisposed) {
+			return;
+		}
 
 		this.layout(width);
 		if (editable) {

